go/tasks/plugins/array/k8s/task.go

```diff
@@ -0,0 +1,231 @@
+	LogLinks         []*idlCore.TaskLog
+	State            *arrayCore.State
+	NewArrayStatus   *arraystatus.ArrayStatus
+	Config           *Config
+	ChildIdx         int
+	MessageCollector *errorcollector.ErrorMessageCollector
+}
+
+type LaunchResult int8
+type MonitorResult int8
+
+const (
+	LaunchSuccess LaunchResult = iota
+	LaunchError
+	LaunchWaiting
+	LaunchReturnState
+)
+
+const (
+	MonitorSuccess MonitorResult = iota
+	MonitorError
+)
+
+func (t Task) Launch(ctx context.Context, tCtx core.TaskExecutionContext, kubeClient core.KubeClient) (LaunchResult, error) {
+	podTemplate, _, err := FlyteArrayJobToK8sPodTemplate(ctx, tCtx)
+	if err != nil {
+		return LaunchError, errors2.Wrapf(ErrBuildPodTemplate, err, "Failed to convert task template to a pod template for a task")
+	}
+	// Remove owner references for remote cluster execution
+	if t.Config.RemoteClusterConfig.Enabled {
+		podTemplate.OwnerReferences = nil
+	}
+	var args []string
+	if len(podTemplate.Spec.Containers) > 0 {
+		args = append(podTemplate.Spec.Containers[0].Command, podTemplate.Spec.Containers[0].Args...)
+		podTemplate.Spec.Containers[0].Command = []string{}
+	} else {
+		return LaunchError, errors2.Wrapf(ErrReplaceCmdTemplate, err, "No containers found in podSpec.")
+	}
+
+	indexStr := strconv.Itoa(t.ChildIdx)
+	podName := formatSubTaskName(ctx, tCtx.TaskExecutionMetadata().GetTaskExecutionID().GetGeneratedName(), indexStr)
+
+	pod := podTemplate.DeepCopy()
+	pod.Name = podName
+	pod.Spec.Containers[0].Env = append(pod.Spec.Containers[0].Env, corev1.EnvVar{
+		Name:  FlyteK8sArrayIndexVarName,
+		Value: indexStr,
+	})
+
+	pod.Spec.Containers[0].Env = append(pod.Spec.Containers[0].Env, arrayJobEnvVars...)
+	pod.Spec.Containers[0].Args, err = utils.ReplaceTemplateCommandArgs(ctx, args, arrayJobInputReader{tCtx.InputReader()}, tCtx.OutputWriter())
+	if err != nil {
+		return LaunchError, errors2.Wrapf(ErrReplaceCmdTemplate, err, "Failed to replace cmd args")
+	}
+
+	pod = ApplyPodPolicies(ctx, t.Config, pod)
+	pod = applyNodeSelectorLabels(ctx, t.Config, pod)
+	pod = applyPodTolerations(ctx, t.Config, pod)
+
+	allocationStatus, err := allocateResource(ctx, tCtx, t.Config, podName)
+	if err != nil {
+		return LaunchError, err
+	}
+	if allocationStatus != core.AllocationStatusGranted {
+		t.NewArrayStatus.Detailed.SetItem(t.ChildIdx, bitarray.Item(core.PhaseWaitingForResources))
+		t.NewArrayStatus.Summary.Inc(core.PhaseWaitingForResources)
+		return LaunchWaiting, nil
+	}
+
+	err = kubeClient.GetClient().Create(ctx, pod)
+	if err != nil && !k8serrors.IsAlreadyExists(err) {
+		if k8serrors.IsForbidden(err) {
+			if strings.Contains(err.Error(), "exceeded quota") {
+				// TODO: Quota errors are retried forever, it would be good to have support for backoff strategy.
+				logger.Infof(ctx, "Failed to launch  job, resource quota exceeded. Err: %v", err)
+				t.State = t.State.SetPhase(arrayCore.PhaseWaitingForResources, 0).SetReason("Not enough resources to launch job")
```

Comment:
is this the same as plugin manager?

Reply:
As in ? yes. If you are referring to same phase? then yes